### PR TITLE
09 get/api/articles/article id/comments

### PR DESCRIPTION
**Pull request by laikaPhoneHome**
This will add the complete get comments with error handling, may contain task 11, I did my best to try and add that to another endpoint.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,57 +1,22 @@
...
+## To successfully connect to the two databases locally:
 
-We will be building an API for the purpose of accessing application data programmatically. The intention here is to mimic the building of a real world backend service (such as reddit) which should provide this information to the front end architecture.
 
-Your database will be PSQL, and you will interact with it using [node-postgres](https://node-postgres.com/).
 
-## Kanban
 
-### Link to your Trello Board here: https://trello.com/b/7yiHe1nI
+Your clone must include the .env files containing PGDATABASE=(nc_news/ nc_news_test)
```
Comment (alexdevdavis) [inline README.md:8]: naturally, in production you'd want to protect your db names, but for the purposes of this project, this is probably ok

File: __tests__/app.test.js [hunk 2]
```diff
@@ -144,6 +165,50 @@ describe('GET', () => {
...
+                describe('/comments',() => {
+                    test('Responds with status 200 and an array of the associated comments, sorted by date with most recent first', () => {
+                        return request(app)
+                        .get('/api/articles/5/comments')
+                        .expect(200)
+                        .then(( {body} ) => {
+                            const { comments } = body;
+
+                            expect(comments).toHaveLength(2);
+                            // expect(comments).toBeSortedBy('created_at', { decending: true}); - dates' format doesnt work
```
Comment (alexdevdavis) [inline __tests__/app.test.js:177]: floaty comment here

File: models/NC-news-models.js [hunk 3]
```diff
@@ -7,36 +7,52 @@ exports.fetchTopics = () => {
...
     let defaultQuery = `
     SELECT articles.*, COUNT(comment_id) AS comment_count
     FROM articles
     LEFT JOIN comments 
     ON articles.article_id = comments.article_id `
 
-    if(topic){
-        defaultQuery += ` WHERE articles.topic = $1 `
-    }
+    topic ?
```
Comment (alexdevdavis) [inline models/NC-news-models.js:28]: would this be DRYer to save query as a string, have a conditional check for topic with a string +=, then following group and order string +=
...as you're turn it into a string to use anyway, unsure of the benefit to using an array for this

**Review by alexdevdavis — APPROVED** [review]
Nicely done! PR approved ... just a refactor suggestion from me for your conditional string query

File: README.md (hunk 1 above)
Comment (laikaPhoneHome) [reply README.md:8]: Thanks alex, noted

File: __tests__/app.test.js (hunk 2 above)
Comment (laikaPhoneHome) [reply __tests__/app.test.js:177]: Yeah sorry ill axe that now

File: models/NC-news-models.js (hunk 3 above)
Comment (laikaPhoneHome) [reply models/NC-news-models.js:28]: ah because I got confused I thought we would be adding more queries or something later so I made it easier to do so😂

**Pull request merged**